### ISSUE TITLE
Tokenize a string with a regular expression.

### ISSUE THREAD

**Issue by xadupre** [Microsoft/onnxruntime#218]
Tokenizer uses separators to tokenizer a string, scikit-learn is using regular expressions. 
Possible implementation: add an option to use regular expression to tokenize. See https://github.com/onnx/onnxmltools/pull/204.

**Comment by wschin** [comment]
The default [regular expression](https://docs.python.org/3/library/re.html) is **r"(?u)\b\w\w+\b"**, which can be achieved by putting **"\u"** and **"\b"** to the separator list of the current tokenizer. I am not sure how popular customized regular expression is, but I never touched that attribute by myself. In addition, I am not confident that supporting regexp is good to runtime because it has no standard and varies from library to library. Could we still use native separator list to mimic **r"(?u)\b\w\w+\b"** until seeing some strong feedbacks from users? @yuanbyu, do you have any comments here?

**Comment by xadupre** [comment]
I'm in favor of using C++ regex. It is easier to understand from a user point of view and this need might come back in the future.

**Comment by yuanbyu** [comment]
I don't have a strong opinion here. I assume that we don't plan to change the current support of separator list, and regular expression could be added later without breaking backward compatibility.  I also assume that we don't have a real use case for regex and this is something forward looking and nice to have.  If this is the case, I wouldn't waste time on this for now.

**Comment by xadupre** [comment]
This change involves two libraries: onnxruntime and scikit-onnx which holds the converter. I'd prefer to do this change only once.

**Comment by wschin** [comment]
Changing converter is easier than changing runtime. As runtime already has an implementation, could we go with what runtime has for now? Then, customers who want to try tokenizer will be able to provide feedback which can guide us.

**Comment by WilliamTambellini** [comment]
Hum, I guess the StringNormalizer op would nt help : 
https://github.com/onnx/onnx/blob/master/docs/Operators.md#stringnormalizer
Should nt tokenization ("string splitting" ? "StringTokenization" op?) be simply added to onnx standard ?

**Comment by yuslepukhin** [comment]
https://github.com/Microsoft/onnxruntime/pull/480
https://github.com/Microsoft/onnxruntime/pull/617
